packages/main/src/plugin/docker/docker-compatibility.ts

```diff
@@ -46,4 +60,106 @@ export class DockerCompatibility {
 
     this.#configurationRegistry.registerConfigurations([dockerCompatibilityConfiguration]);
   }
+
+  protected getTypeFromServerInfo(info: { OperatingSystem?: string }, podmanInfo: unknown): DockerSocketServerInfoType {
+    if (info.OperatingSystem === 'Docker Desktop') {
+      return 'docker';
+    } else if (info.OperatingSystem === 'podman' || podmanInfo) {
+      // if podman info is available, then it is podman
+      return 'podman';
+    }
+    return 'unknown';
+  }
+
+  async resolveLinkIfAny(path: string): Promise<string> {
+    // do not resolve on Windows
+    if (isWindows()) {
+      return path;
+    }
+
+    const lstat = await promises.lstat(path);
+    if (lstat.isSymbolicLink()) {
+      const resolved = await promises.readlink(path);
+      return this.resolveLinkIfAny(resolved);
+    }
+
+    return path;
+  }
+
+  async getSystemDockerSocketMappingStatus(): Promise<DockerSocketMappingStatusInfo> {
+    // first, try to connect to the docker socket if on Linux or macOS
+    // if the connection is successful, return @the status as running
+
+    const socketPath = isWindows() ? DockerCompatibility.WINDOWS_NPIPE : DockerCompatibility.UNIX_SOCKET_PATH;
+
+    const dockerode = new Dockerode({ socketPath });
+    // check if we can also use a podman API
+    const libpodDockerode = dockerode as unknown as LibPod;
+
+    try {
+      const compatInfo = await dockerode.info();
+      let podmanInfo = undefined;
+
+      let serverVersion = compatInfo.ServerVersion;
+      try {
+        podmanInfo = await libpodDockerode.podmanInfo();
+        serverVersion = podmanInfo.version.Version;
+      } catch (error: unknown) {
+        // podman API is not available
+      }
+
+      const status: DockerSocketMappingStatusInfo = {
+        status: 'running',
+        serverInfo: {
+          type: this.getTypeFromServerInfo(compatInfo, podmanInfo),
+          serverVersion,
+          operatingSystem: compatInfo.OperatingSystem,
+          osType: compatInfo.OSType,
+          architecture: compatInfo.Architecture,
+        },
+      };
+
+      // now, try to see if the socket path is an alias to a podman machine
+      if (isMac()) {
+        // is UNIX_SOCKET_PATH a symbolic link ?
+        const socketPath = await this.resolveLinkIfAny(DockerCompatibility.UNIX_SOCKET_PATH);
+
+        // do we have a matching provider for the socket path ?
+        const currentConnections = this.#providerRegistry.getContainerConnections();
+
+        // search if we have a connection with the same socket path
+        const foundConnection = currentConnections.find(c => c.connection.endpoint.socketPath === socketPath);
+
+        // provider is
```

Comment:
is it complete?

Reply:
I think I moved to another task 😂

I will check what I wanted to say

Reply:
fixed